expo/tests/gbe/test_assign_volunteer.py

```diff
@@ -0,0 +1,275 @@
+import gbe.models as conf
+import nose.tools as nt
+from unittest import TestCase
+from django.test.client import RequestFactory
+from django.test import Client
+from django.core.exceptions import PermissionDenied
+from datetime import datetime, date, time
+import pytz
+import re
+from gbe.views import assign_volunteer
+from django.contrib.auth.models import Group
+from tests.factories.gbe_factories import (
+    PersonaFactory,
+    ProfileFactory,
+    VolunteerFactory,
+    ConferenceFactory,
+    GenericEventFactory,
+    UserFactory,
+    VolunteerWindowFactory,
+)
+from tests.functions.gbe_functions import login_as
+from scheduler.models import Event as sEvent
+from scheduler.models import (
+    Worker,
+    ResourceAllocation,
+    EventContainer
+)
+
+
+class TestAssignVolunteer(TestCase):
+    '''Tests for review_volunteer view'''
+
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.client = Client()
+        self.performer = PersonaFactory.create()
+        self.privileged_profile = ProfileFactory.create()
+        self.privileged_user = self.privileged_profile.user_object
+        group, nil = Group.objects.get_or_create(name='Volunteer Coordinator')
+        self.privileged_user.groups.add(group)
+
+    @nt.raises(PermissionDenied)
+    def test_assign_volunteers_bad_user(self):
+        ''' user does not have Volunteer Coordinator, permission is denied'''
+        volunteer = VolunteerFactory.create()
+        request = self.factory.get('volunteer/assign/%d' % volunteer.pk)
+        request.user = ProfileFactory.create().user_object
+        request.session = {'cms_admin_site': 1}
+        login_as(request.user, self)
+        response = assign_volunteer(request, volunteer.pk)
+
+    @nt.raises(PermissionDenied)
+    def test_assign_volunteers_no_profile(self):
+        ''' user does not have a profile, permission is denied'''
+        volunteer = VolunteerFactory.create()
+        request = self.factory.get('volunteer/assign/%d' % volunteer.pk)
+        request.user = UserFactory.create()
+        request.session = {'cms_admin_site': 1}
+        login_as(request.user, self)
+        response = assign_volunteer(request, volunteer.pk)
+
+    def test_assign_volunteer_no_events(self):
+        ''' assign a volunteer when there are no opportunities'''
+        volunteer = VolunteerFactory.create()
+        request = self.factory.get('volunteer/assign/%d' % volunteer.pk)
+        request.user = self.privileged_user
+        request.session = {'cms_admin_site': 1}
+        login_as(request.user, self)
+        response = assign_volunteer(request, volunteer.pk)
+        nt.assert_equal(response.status_code, 200)
+        nt.assert_true('Assign Volunteer to Opportunities' in response.content)
+
+    def test_assign_volunteer_show_current_events(self):
```

Comment:
A test should test one thing - this test tests many things. This is not a blocker, since we need to get this code merged, but this will need to be broken up into a suite of tests.